src/redux/slices/postSlice.js

```diff
@@ -0,0 +1,194 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const initialState = {
+  status: "idle",
+  allPosts: [],
+  allUsers: [],
+  postId: null,
+  error: null,
+  foundUser: {},
+  isEdit: false,
+  content: "",
+};
+
+export const getAllPosts = createAsyncThunk("posts/getAllPosts", () => {
+  return (async () => {
+    try {
+      const response = await axios({
+        method: "get",
+        url: `/api/posts/`,
+      });
+      return response.data.posts;
+    } catch (error) {
+      return error;
+    }
+  })();
+});
+
+// This will go to userSlice later
+export const getAllUsers = createAsyncThunk("posts/getAllUsers", () => {
+  return (async () => {
+    try {
+      const response = await axios({
+        method: "get",
+        url: "/api/users",
+      });
+      return response.data.users;
+    } catch (error) {
+      return error;
+    }
+  })();
+});
+
+// This will go to authSlice later
+export const logIn = createAsyncThunk("posts/logIn", () => {
+  return (async () => {
+    try {
+      const response = await axios({
+        method: "POST",
+        url: "/api/auth/login",
+        data: { username: "adarshbalika", password: "adarshBalika123" },
+      });
```

Comment:
suggestion: I guess you need to add this conditionally, otherwise every time logs in with the same login credentials

Reply:
yes, in the next pr I am implementing authentication there I will do this.